doc/developer/feature-lifecycle.md

```diff
@@ -0,0 +1,111 @@
+
+## Private Preview
+
+The goal of Private Preview is to identify and address all of
+the reasons why the implementation of a feature might not work.
+To do this, the DRI is responsible for getting the feature in
+front of internal and external testers by posting in the
+#epd-announce channel.
+
+Testers may raise a few different classes of issues:
+[value, usability, feasibility, and viability](https://www.svpg.com/four-big-risks/).
+Some of these issues may be significant enough for the DRI to
+reconsider the chosen solution, effectively sending a project
+back to the Discovery phase. Other issues may be smaller, and
+fixes can be made within the scoped solution. The Private Preview
+phase may take a few days for smaller features, and up to a few
+weeks for larger ones.
+
+_Note: features in Private Preview will not be available in
+customer environments by default, they will be turned on for
+individual customers via the
+[limit increase process](https://www.notion.so/Adjusting-environment-limits-ba79e95871734ab2a292ecc8698d01e9).
+Customers who test these features should be warned that they
+are subject to significant, and occasionally incompatible, changes._
+
+Before a feature can move to Public Preview, it must:
+
+- Be verified that it solves the targeted problem for at least
+  one external tester, with additional optional internal testers.
+- Be entirely code complete, if it wasn’t already.
+- Have comprehensive tests live and running.
+- Have basic observability live and running.
+- Have basic metrics live and running.
+- Have addressed all known stability issues.
+- Have addressed all known limitations, except those that are explicitly out of scope.
+- Support all known customer configurations, except those that are explicitly out of scope.
+- Have polished reference documentation live behind a {{< public-preview >}} flag.
+- Have a merged release note behind a {{< public-preview >}} flag.
+
+## Public Preview
+
+The goal of Public Preview is to ensure the stability of a
+feature. In this phase, features are available in all customer
+environments by default.
+
+The Public Preview phase is likely to take the longest to
+complete: it depends on internal confidence in the stability of
+a feature as well as its adoption by our users. During this phase,
+the DRI may continue to collect feedback about the feature. The
+DRI is responsible for curating that feedback and applying
+changes as needed.
+
+Before a feature can move to General Availability, it must:
+
+- Be used by at least two customers in production use cases.
+- Have additional documentation (like user guides) live, or
+  an explicit justification for why it’s not required.
+- Have a blog post describing the feature live, or an explicit
+  justification for why it’s not required.
+- Have a merged release note that the feature is in General Availability.
+- Have a merged Changelog announcement.
+
+## General Availability
```

Comment:
Should we discuss at this stage whether the feature flag shall stay or be removed?

Reply:
Added in c358e76750.